### ISSUE TITLE
Data Matrix and Aztec codes scan not working

### ISSUE THREAD

**Issue by MaratEldarov** [HMS-Core/hms-cordova-plugin#22]
**Description**
I have a problem with Data Matrix and Aztec code scan, but simple QR code works fine.

**Expected behavior**
I expect that Aztec and Data Matrix scanning will work fine and give me the result when I pass `HMSScan.ScanTypes.ALL_SCAN_TYPE` in `scanTypes` array:

```
const defaultViewModeInput = {
  scanTypes: [window.HMSScan.ScanTypes.ALL_SCAN_TYPE],
};

await window.HMSScan.defaultViewMode(defaultViewModeInput);
```

**Current behavior**
But when I scan Aztec or Data Matrix codes - scanner not react on it and staying opened. When I try scan that codes from Gallery - I got error:

![image](https://user-images.githubusercontent.com/35111740/109956609-44562b00-7cfd-11eb-91d6-0a6f7552c441.png)

In English it means something like "code not defined"

**Environment**
 - Platform: Cordova 8.1.2, Cordova Android 8.0.0
 - Kit: Scan Kit
 - Kit Version 1.2.3-300
 - OS Version Android 11

**Other**
These are codes I try to scan:

![image](https://user-images.githubusercontent.com/35111740/109957984-212c7b00-7cff-11eb-80df-0e41ae71a1f9.png)

![image](https://user-images.githubusercontent.com/35111740/109958001-27225c00-7cff-11eb-8c52-e934484a3890.png)

**Comment by MaratEldarov** [comment]
I also using forked version of plugin form here https://github.com/cempay/hms-cordova-plugin
because of there was some little fixes from our team. Without them we have problems with getting permissions.

And here is my full JS code

```
async function requestPermissionsScan() {
  try {
    await window.HMSScan.requestPermissions({
      permissionList: ['CAMERA', 'WRITE_EXTERNAL_STORAGE'],
    });
  } catch (ex) {
    throw new Error(ex);
  }
}

async function checkPermissions() {
  try {
    const result = await window.HMSScan.checkPermissions({
      permissionList: ['CAMERA', 'WRITE_EXTERNAL_STORAGE'],
    });
    return result;
  } catch (ex) {
    console.error('checkPermissions error:', { ex });
  }
}

async function defaultViewMode() {
  try {
    const defaultViewModeInput = {
      scanTypes: [window.HMSScan.ScanTypes.ALL_SCAN_TYPE],
    };
    const { originalValue: text } = await window.HMSScan.defaultViewMode(defaultViewModeInput);
    return { text };
  } catch (ex) {
    throw new Error(ex);
  }
}

async function scanHuawei() {
  if (!window.HMSScan) {
    throw new Error('HMS BarcodeScanner plugin not exists!');
  }
  const { CAMERA, WRITE_EXTERNAL_STORAGE } = await checkPermissions();
  if (!CAMERA.hasPermission || !WRITE_EXTERNAL_STORAGE.hasPermission) {
    await requestPermissionsScan();
  }
  try {
    const scanResult = await defaultViewMode();
    return scanResult;
  } catch (scannerException) {
    throw new Error(scannerException.message);
  }
}
```

**Comment by furkanaybasti** [comment]
Hello MaratEldarov, we are investigating the error you encountered, we will return as soon as possible.

**Comment by furkanaybasti** [comment]
If you update your Scan SDK version to 1.3.1.300, the problem you encounter will be solved.

1. In the plugin.xml file in the plugin **<framework src = "com.huawei.hms: scan: 1.2.5.300" />** you need to change this line **<framework src = "com.huawei.hms: scan: 1.3.1.300" />** this way.
2. In Plugin, you need to change the implementation information in this directory.  **src-> android-> build.gradle** implementation 'com.huawei.hms: scan: 1.3.1.300'

**Comment by MaratEldarov** [comment]
I try to do it and got plenty of errors about android.support packages in other plugins, for example:

![image](https://user-images.githubusercontent.com/35111740/111167305-692e8600-85ba-11eb-8699-afd2fe1102bb.png)

![image](https://user-images.githubusercontent.com/35111740/111167326-6df33a00-85ba-11eb-8629-fc16e3a56e5c.png)

![image](https://user-images.githubusercontent.com/35111740/111167348-76e40b80-85ba-11eb-8646-989a6f59c186.png)

![image](https://user-images.githubusercontent.com/35111740/111167362-7b102900-85ba-11eb-9260-0013965ce5d7.png)

![image](https://user-images.githubusercontent.com/35111740/111167387-819ea080-85ba-11eb-9cbe-59649b87f4d4.png)

**Comment by furkanaybasti** [comment]
I am developing the new version of Scankit.  It will be released at the end of next month.

**Comment by aktug** [comment]
https://www.npmjs.com/package/@hmscore/cordova-plugin-hms-scan

**Comment by aktug** [comment]
Released v1.3.1-300

https://www.npmjs.com/package/@hmscore/cordova-plugin-hms-scan
https://www.npmjs.com/package/@hmscore/ionic-native-hms-scan
